handlers/server_test.go

```diff
@@ -0,0 +1,615 @@
+func (m *MockDynamoDB) GetItem(_ *dynamodb.GetItemInput) (*dynamodb.GetItemOutput, error) {
+	return &m.GetOutput, m.GetError
+}
+
+func (m *MockDynamoDB) UpdateItem(_ *dynamodb.UpdateItemInput) (*dynamodb.UpdateItemOutput, error) {
+	return &m.UpdateOutput, m.UpdateError
+}
+
+func (m *MockDynamoDB) DeleteItem(_ *dynamodb.DeleteItemInput) (*dynamodb.DeleteItemOutput, error) {
+	return &m.DeleteOutput, m.DeleteError
+}
+
+func (m *MockDynamoDB) PutItem(_ *dynamodb.PutItemInput) (*dynamodb.PutItemOutput, error) {
+	return &m.PatchOutput, m.PatchError
+}
+
+func TestServer_GetEmployee(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Server handler")
+}
+
+var _ = Describe("Employee CRUD Server tests", func() {
+	var (
+		server Server
+		m      MockDynamoDB
+	)
+
+	BeforeEach(func() {
+		m = MockDynamoDB{}
+		server.SetupRouter()
+		server.Db = &m
+	})
+
+	Context("Happy Path tests", func() {
+		It("Status:200 Get Employee Success", func() {
+
+			req, err := http.NewRequest("GET", "/v1/employee/{id}", nil)
+			if err != nil {
+				fmt.Printf("could not create a request: %v", err.Error())
+			}
+
+			w := httptest.NewRecorder()
+
+			//some fake UUID
+			vars := map[string]string{"id": "663c9932-383d-44d1-8b9d-1071731a6312"}
+
+			req = mux.SetURLVars(req, vars)
+
+			newEmployee := store.Employee{
+				Name:       "shayan",
+				Email:      "shayan@twilio.com",
+				EmployeeId: "1",
+				City:       "Yar",
+				Address:    "The Black Pearl",
+				Department: "Piratology",
+			}
+
+			item, _ := dynamodbattribute.MarshalMap(newEmployee)
+
+			m.GetError = nil
+			m.GetOutput = dynamodb.GetItemOutput{
+				Item: item,
+			}
+
+			server.GetEmployee(w, req)
+
+			res := w.Result()
+			defer res.Body.Close()
+
+			var resp GetEmployeeResponse
+
+			err = json.NewDecoder(res.Body).Decode(&resp)
+			if err != nil {
+				fmt.Printf("unable to successfully decode response got error %v", err)
+			}
+
+			Expect(resp.Status).To(Equal(http.StatusOK))
+			Expect(resp.Data).To(Equal(newEmployee))
+		})
+		It("Status:201 Post Employee Success", func() {
+			reader := strings.NewReader(`{"employeeName": "pablo", "city":  "Denver", "address": "3 Main St", "department": "eng"}`)
+
+			req, err := http.NewRequest("POST", "/v1/employee/add", reader)
+			if err != nil {
+				fmt.Printf("could not create a request: %v", err.Error())
+			}
+
+			w := httptest.NewRecorder()
+
+			server.PostEmployee(w, req)
+
+			res := w.Result()
+			defer res.Body.Close()
+
+			var resp GetOneEmployee
+
+			err = json.NewDecoder(res.Body).Decode(&resp)
+			if err != nil {
+				fmt.Printf("unable to successfully decode response got error %v", err)
+			}
+
+			Expect(resp.Status).To(Equal(http.StatusCreated))
+			Expect(resp.Employee.Name).To(Equal("pablo"))
+			Expect(resp.Employee.Email).To(Equal("pablo@twilio.com"))
+			//UUID REGexp
+			r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
```

Comment:
this is good

but you could also use `uuid.Parse()` or something similar to save yourself the pain of regex.

Reply:
it would return an error if invalid and you can `Expect()` it to be nil.